mne/viz/_3d.py

```diff
@@ -1227,19 +1230,29 @@ def _plot_sensors(renderer, info, to_cf_t, picks, meg, eeg, fnirs,
         if len(locs[sensor_type]) > 0:
             sens_loc = np.array(locs[sensor_type])
             sens_loc = sens_loc[~np.isnan(sens_loc).any(axis=1)]
-            color = defaults[sensor_type + '_color']
             scale = defaults[sensor_type + '_scale']
-            actor, _ = _plot_glyphs(renderer=renderer, loc=sens_loc * scalar,
-                                    color=color, scale=scale * scalar,
-                                    opacity=sensor_opacity,
-                                    orient_glyphs=orient_glyphs,
-                                    scale_by_distance=scale_by_distance,
-                                    project_points=project_points,
-                                    surf=surf, check_inside=check_inside,
-                                    nearest=nearest)
+            actor_list = []
+            for idx_sen in range(len(info["ch_names"])):
+                if sensor_colors is None:
+                    color = defaults[sensor_type + '_color']
+                else:
+                    if type(sensor_colors) == list:
+                        color = sensor_colors[idx_sen]
+                    else:
+                        color = sensor_colors[idx_sen, :]
+                actor, _ = _plot_glyphs(renderer=renderer,
+                                        loc=(sens_loc * scalar)[idx_sen, :],
```

Comment:
some tests are failing with `IndexError` on this line. Can you figure out why?  Offhand guess is that you're not accounting for bad channels, but that's just a guess.

Reply:
@drammock It would be helpful to know how to run the specific test. E.g. I see that the test `test_notebook_alignment()` in `mne.viz._brain.tests` fails due to an index error. How can I run this single test?

And is there optimally also an option for debugging within this specific test? When I put a breakpoint in VSCode in the specific line and try to debug, it won't jump to the breakpoint and does not seem to debug it correctly.

I tried to run `pytest mne/viz/_brain/tests/test_notebook.py` but it just skipped the test since it "Requires testing data". 

Sorry for not being so proficient about using the testing framework. But I am very happy to learn more about it.

Reply:
Ok, I read the documentation and installed the testing data. It is for me nevertheless confusing that I could not debug within the test:
![image](https://user-images.githubusercontent.com/38216460/198158435-acc6a287-24a0-46e2-a98c-1c95215ecd04.png)

So I can run the tests now, but VSCode does not stop at the breakpoint.
I tried to set the launch.json accordingly https://stackoverflow.com/a/60678160 and read through a related issue https://github.com/microsoft/vscode-python/issues/693 but was not able to fix it.

Debugging would be probably quite necessary for this type of issue.

Reply:
to run just one test and automatically enter debugger on error you can do `pytest --pdb path/to/test/file.py -k name_of_test_function`